metadata-ingestion/src/datahub/ingestion/api/source_helpers.py

```diff
@@ -274,6 +279,56 @@ def auto_browse_path_v2(
         telemetry.telemetry_instance.ping("incorrect_browse_path_v2", properties)
 
 
+def auto_empty_dataset_usage_statistics(
+    stream: Iterable[MetadataWorkUnit],
+    *,
+    dataset_urns: Set[str],
+    config: BaseTimeWindowConfig,
+    all_buckets: bool = False,  # TODO: Enable when CREATE changeType is supported for timeseries aspects
+) -> Iterable[MetadataWorkUnit]:
+    """Emit empty usage statistics aspect for all dataset_urns ingested with no usage."""
+    buckets = config.buckets() if all_buckets else config.majority_buckets()
+
+    # Maps time bucket -> urns with usage statistics for that bucket
+    usage_statistics_urns: Dict[int, Set[str]] = {bucket: set() for bucket in buckets}
+
+    for wu in stream:
+        yield wu
+        if not wu.is_primary_source:
+            continue
+
+        urn = wu.get_urn()
+        if guess_entity_type(urn) == DatasetUrn.ENTITY_TYPE:
+            dataset_urns.add(urn)
```

Comment:
Looks like we are already passing a well-populated set of `dataset_urns` to this helper function from our sources. This line again adds all the dataset urns - kind of redundant but a safety net against what connector may have missed ?

Reply:
Ah this was left in back when I was not passing them in (before I realized this was not guaranteed, e.g. when `include_technical_schema` is False for snowflake). I think it's pretty harmless to keep and safety net feature may be nice